events.go

```diff
@@ -133,6 +133,9 @@ func (s *Session) handleNodeEvent(frames []frame) {
 		// TODO: can we be sure the order of events in the buffer is correct?
 		switch f := frame.(type) {
 		case *topologyChangeEventFrame:
+			host, port := s.cfg.translateAddress(f.host.String(), f.port)
```

Comment:
So ... in looking into your suggestion, I'm thinking it might actually just be easiest to avoid processing and storing the translated address anywhere. Instead, it might be easiest to simply be best to translate in `Session.connect` before calling `Connect`. I'll add some debugging in that method, too.